pkg/controller/jobframework/tas.go

```diff
@@ -73,7 +73,7 @@ func (p *podSetTopologyRequestBuilder) Build() (*kueue.PodSetTopologyRequest, er
 		}
 		psTopologyReq.Unconstrained = &unconstrained
 	default:
-		if !sliceRequiredTopologyFound || !sliceSizeFound {
+		if (!sliceRequiredTopologyFound || !sliceSizeFound) && (p.podIndexLabel == nil && p.subGroupIndexLabel == nil && p.subGroupCount == nil) {
```

Comment:
Oh, we do not have any UTs for the jobframework tas.go.
We should implement that as a follow up.

Reply:
We don't have for the framework itself but we do have them for particular integrations, e.g.: https://github.com/kubernetes-sigs/kueue/pull/6615/files#diff-d7e3e8f98ff2760e6e7d9f956ce420f7cfa4144cecbaa484c036fbf9cc686e4aR285

Those tests check the `PodSets` which then directly calls functions from this file

Reply:
Yes, but ideally, it would be better to have cases in jobframework/tas.go since this is a unit test.

Reply:
I agree